### ISSUE TITLE
[BUG] https://www.wowhead.com/cata/spell=22842/frenzied-regeneration

### ISSUE THREAD

**Issue by Gunda456** [JamminL/cata-classic-bugs#1265]
### 1. Current Behavior:
it increase hp to 24% if below that value 
### 2 How to reproduce
      go below 24% hp and use regen
  #### 1.3  Source Material
https://youtu.be/xrirKXPv30g?si=EzGp2NPfXkikzznR&t=29
{Describe how that mechanic/spell/npc should be working, in detail}
it should increase health to 30% (if below that value) as it says in the tooltip
https://www.wowhead.com/cata/spell=22842/frenzied-regeneration
{Video, Posts, etc..}
https://youtu.be/-4hHCiGHTAI?si=TtV5Q7Wl-nQ9r12y&t=298

**Comment by AidanZMoon** [comment]
This is accurate behavior to 4.3.4
The way this works is it IMMEDIATELY grants you the health % if you are below 30%, and THEN it gives you 30% maximum health.
So you are put at 30% of your OLD health value BEFORE the 30% increase.
So you are placed at 23% of your CURRENT health, because that is 30% of your old Health.
Your video from 12 years ago also demonstrates this so I'll use the numbers from it:
The Druid's Maximum health right before pressing Frenzied Regeneration is 184.5k in bear form.
Their Maximum health changes to 239.9k. Which is a 184.5 x 1.3 = 239.9 so that lines up.
But his Current health value that he is set to with Frenzied Regeneration is 56522. He takes a tick of healing for 1164 right then so he is really set to 56522 - 1164 = 55358

If this regen was based off of the value of his new maximum health you would expect 239.9 * 0.3 = 71.97
But it is based off of his Old health which was 184.5 * 0.3 = 55.35 which lines up exactly with what you see in the 12 year old.

TL;DR Frenzied Regeneration sets your health when you are below 30% to 30% of your Pre-Frenzied Regeneration boosted health pool, not Post-Frenzied Regeneration boosted health.
So you are set to 30% of your old health, or 23% of your new total after the boost.

This is accurate to 4.3.4 Cataclysm.
